adapter/redisquota/redisquota.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package redisquota provides a quota implementation with redis as backend.
+// The prerequisite is to have a redis server running.
+package redisquota
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	ptypes "github.com/gogo/protobuf/types"
+
+	"istio.io/mixer/adapter/redisquota/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/pool"
+)
+
+type builder struct{ adapter.DefaultBuilder }
+
+type redisQuota struct {
```

Comment:
I think creating a mock redis would be useful. I don't know redis, so how hard would this be?

We want to use redis for our config value store for the mixer as a whole. So having a mock redis we could use during testing would be generally useful. @mandarjog 

Probably should check to see if someone's already done a mock redis implementation in Go.